worksheet.rb

```diff
@@ -1,34 +1,165 @@
-# Which layers are nested in each other?
-# Which layers of data "have" within it a different layer?
-# Which layers are "next" to each other?
+all_drivers = [
+		{
+				driver_id: "DR0001",
+				rides: [
+						{
+								rider_id: "RD0003",
+								date: "2016-02-03",
+								cost: 10,
+								rating: 3,
+						},
+						{
+								rider_id: "RD0015",
+								date: "2016-02-03",
+								cost: 30,
+								rating: 4,
+						},
+						{
+								rider_id: "RD0003",
+								date: "2016-02-05",
+								cost: 45,
+								rating: 2,
+						}
+				]
+		},
 
-########################################################
-# Step 2: Assign a data structure to each layer
+		{
+				driver_id: "DR0002",
+				rides: [
+						{
+								rider_id: "RD0073",
+								date: "2016-02-03",
+								cost: 25,
+								rating: 5,
+						},
+						{
+								rider_id: "RD0013",
+								date: "2016-02-04",
+								cost: 15,
+								rating: 1,
+						},
+						{
+								rider_id: "RD0066",
+								date: "2016-02-05",
+								cost: 35,
+								rating: 3,
+						}
+				]
+		},
 
-# Copy your list from above, and in this section
-# determine what data structure each layer should have
-
-########################################################
-# Step 3: Make the data structure!
+		{
+				driver_id: "DR0003",
+				rides: [
+						{
+								rider_id: "RD0066",
+								date: "2016-02-04",
+								cost: 5,
+								rating: 5,
+						},
+						{
+								rider_id: "RD0003",
+								date: "2016-02-05",
+								cost: 50,
+								rating: 2,
+						}
+				]
+		},
 
-# Setup the entire data structure:
-# based off of the notes you have above, create the
-# and manually write in data presented in rides.csv
-# You should be copying and pasting the literal data
-# into this data structure, such as "DR0004"
-# and "3rd Feb 2016" and "RD0022"
+		{
+				driver_id: "DR0004",
+				rides: [
+						{
+								rider_id: "RD0022",
+								date: "2016-02-03",
+								cost: 5,
+								rating: 5,
+						},
+						{
+								rider_id: "RD0022",
+								date: "2016-02-04",
+								cost: 10,
+								rating: 4,
+						},
+						{
+								rider_id: "RD0073",
+								date: "2016-02-05",
+								cost: 20,
+								rating: 5,
+						}
+				]
+		}
+]
 
 ########################################################
 # Step 4: Total Driver's Earnings and Number of Rides
-
 # Use an iteration blocks to print the following answers:
 # - the number of rides each driver has given
+# all_drivers.each do |driver|
+#   puts driver[:driver_id] + " has given" + " " + driver[:rides].length.to_s + " " + "rides."
+# end
+
 # - the total amount of money each driver has made
+# all_drivers.each do |driver|
+#   total_money_made = 0
+#   driver_id = driver[:driver_id]
+#   # p driver
+#
+#   driver[:rides].each do |num|
+#   ride_cost = num[:cost]
+#   total_money_made += ride_cost
+#   end
+#   puts "#{driver_id} earned a total of: $#{total_money_made}."
+# end
```

Comment:
👍 , check your indentation here and I suggest using either the `.sum_by` or `.reduce` methods to calculate the totals.